src/hb_store_gateway.erl

```diff
@@ -215,22 +216,47 @@ cache_read_message_test() ->
 %% know that the module is not respecting the route list.
 specific_route_test() ->
     hb_http_server:start_node(#{}),
+    %% Define the response we want
+    ID = <<"BOogk_XAI3bvNWnxNxwxmvOfglZt17o4MOVAdPNZ_ew">>,
+    Data = <<"specific_route_testx">>,
+    DefaultResponse = {200, Data},
+    Endpoints = [{<<"/raw/", ID/binary>>, ok, DefaultResponse}],
+    {ok, MockServer, _ServerHandle} = hb_mock_server:start(Endpoints),
+    %% Define configuration, we use a valid gateway to obtain a valid response
+    %% and then mock the raw endpoint to our mockserver.
     Opts = #{
         store =>
             [
                 #{ <<"store-module">> => hb_store_gateway, 
-                   <<"routes">> => [],
-                   <<"only">> => local
+                   <<"routes">> => [
+                    #{
+                        <<"template">> => <<"/graphql">>,
+                        <<"nodes">> => [
+                            #{
+                                <<"prefix">> => <<"https://arweave-search.goldsky.com">>,
+                                <<"opts">> => #{
+                                    <<"http_client">> => httpc,
+                                    <<"protocol">> => http2
+                                }
+                            }
+                        ]
+                    },
+                    #{
+                        <<"template">> => <<"/raw">>,
+                        <<"node">> =>
+                            #{
+                                <<"prefix">> => MockServer,
+                                <<"opts">> => #{
+                                    <<"http_client">> => gun,
+                                    <<"protocol">> => http2 
+                                }
+                            }
+                     }
+                   ]
                 }
             ]
     },
-    ?assertMatch(
-        not_found,
-        hb_cache:read(
-            <<"BOogk_XAI3bvNWnxNxwxmvOfglZt17o4MOVAdPNZ_ew">>,
-            Opts
-        )
-    ).
```

Comment:
There is a lot of error behaviours that mimic a "true" `not_found` output. Changed this to a value we expect.